plugins/wasi_llm/wasillmmodule.cpp

```diff
@@ -8,13 +8,15 @@ namespace WasmEdge {
 namespace Host {
 
 WasiLLMModule::WasiLLMModule() : ModuleInstance("wasi_llm") {
-  addHostFunc("model_create", std::make_unique<WasiLLMModelCreate>());
-  addHostFunc("model_free", std::make_unique<WasiLLMModelFree>());
-  addHostFunc("dataloader_create", std::make_unique<WasiLLMDataLoaderCreate>());
-  addHostFunc("dataloader_free", std::make_unique<WasiLLMDataLoaderFree>());
-  addHostFunc("tokenizer_create", std::make_unique<WasiLLMTokenizerCreate>());
-  addHostFunc("tokenizer_free", std::make_unique<WasiLLMTokenizerFree>());
-  addHostFunc("model_train", std::make_unique<WasiLLMModelTrain>());
+  addHostFunc("model_create", std::make_unique<WasiLLMModelCreate>(Env));
+  addHostFunc("model_free", std::make_unique<WasiLLMModelFree>(Env));
```

Comment:
i feel like we don't really need to expose these free functions to the users. since the resources are managed by env object, maybe they can just be freed using an raii style? @hydai

Reply:
Sounds great. We can hide these details inside the `Env` object.